server/src/main/java/com/defold/extender/Extender.java

```diff
@@ -862,51 +862,77 @@ else if (platform.contains("osx")) {
         mergedContextWithPodsForObjC.put("systemIncludes", new ArrayList<String>());
         mergedContextWithPodsForObjCpp.put("systemIncludes", new ArrayList<String>());
         mergedContextWithPodsForSwift.put("systemIncludes", new ArrayList<String>());
-
+        // also clearing systemIncludes for c
+        // see https://github.com/defold/extender/issues/389
+        mergedContextWithPodsForC.put("systemIncludes", new ArrayList<String>());
 
         List<String> objs = new ArrayList<>();
 
         if (!pod.swiftSourceFiles.isEmpty()) {
+            // Add -enable-experimental-feature AccessLevelOnImport
+            // There are a lot of extensions with Swift code that seem to require this
+            // flag. Perhaps swiftFlags should be added as a new field in ext.manifest?
+            // https://github.com/swiftlang/swift-evolution/blob/main/proposals/0409-access-level-on-imports.md
+            List<String> swiftFlags = (List<String>)mergedContextWithPodsForSwift.get("swiftFlags");
+            List<String> updatedSwiftFlags = new ArrayList<String>(swiftFlags);
+            updatedSwiftFlags.add("-enable-experimental-feature");
+            updatedSwiftFlags.add("AccessLevelOnImport");
+            mergedContextWithPodsForSwift.put("swiftFlags", updatedSwiftFlags);
+
             // generate headers from swift files
             List<String> emitSwiftHeaderCommands = new ArrayList<>();
-            emitSwiftHeaders(pod, mergedContextWithPodsForC, emitSwiftHeaderCommands);
+            LOGGER.info("emit swift header");
+            emitSwiftHeader(pod, mergedContextWithPodsForSwift, emitSwiftHeaderCommands);
```

Comment:
Use the correct context, use mergedContextWithPodsForSwift not mergedContextWithPodsForC